### PR TITLE
Update Pipfile to point cantools package to forked version with no doubles

### PR DESCRIPTION
**Pull request by timbitss**
### Summary
Updated Pipfile to install the cantools package from FormulaE [fork](https://github.com/UBCFormulaElectric/cantools), removing the usage of doubles in `encode()` and `decode()` functions. V33.1.1 is still maintained.

Motivation: Some of the inverter CAN signals have an offset of 0.1 and thus require the encode() and decode() functions. The use of doubles in the autogenerated encode() and decode() functions are unnecessary since it increases the execution time (STM32F4's FPU supports floats only). Therefore, I updated cantools to generate code that uses floats instead of doubles.

Before App_CanMsgs.c:
image

After App_CanMsgs.c:
image

### Changelist 
- Updated Pipfile to install the cantools package from FormulaE [fork](https://github.com/UBCFormulaElectric/cantools). 
- Updated year of stm32f3xx_hal_conf.h file in each board folder from 2021 to 2022

### Testing Done

1. Entered  `pipenv uninstall -all` in the command line to uninstall all packages without modifying the Pipfile.  
2. Entered `pipenv install` to re-install all packages including cantools which is now located at the fork.
3. Built .elf file to see that doubles had changed to floats in App_CanMsgs.h and App_CanMsgs.c

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

**Comment by alxshev** [comment]
What's the rationale for doing this?

**Comment by timbitss** [comment]
> Why is this being done?

Some of the inverter CAN signals have an offset of 0.1 and thus require the encode() and decode() functions. The use of doubles in the autogenerated encode() and decode() functions are unnecessary since it increases the execution time (STM32F4's FPU supports floats only). Therefore, I updated cantools to generate code that uses floats instead of doubles.

Counterpoint: we could go without using the encode() and decode() functions and do the conversions directly, but I think it's more elegant and less error-prone if we use the auto-gen code.

**Review by alxshev — APPROVED** [review]
lgtm, just add what you said in the comments to the PR description so people in the future know why this was done

**Comment by akrantz01** [comment]
It looks like there is already an issue open for adding this to mainline cantools (cantools/cantools#393). I think it would be better to wait a bit for the feature request to get resolved and update to the latest version than maintain a custom fork of cantools. Since if we ever want to update the cantools version, we would need to manually update the fork every time.

If we wanted this sooner, it probably wouldn't be too difficult to adapt your change into a PR for the cantools repo to implement this feature.

File: Pipfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ name = "pypi"
 pylint = "*"
 
 [packages]
-cantools = "~=33.1.1"
+cantools = {git = "git://github.com/timbitss/cantools.git"}
```
Comment (pasitlaothamatas) [inline Pipfile:10]: if we decide to go with the forking approach, i think its a better idea to fork into UBCFormulaElectric org and update the following line to `{git = "git://github.com/UBCFormulaElectric/cantools.git"}`

**Comment by pasitlaothamatas** [comment]
> It looks like there is already an issue open for adding this to mainline cantools ([cantools/cantools#393](https://github.com/cantools/cantools/issues/393)). I think it would be better to wait a bit for the feature request to get resolved and update to the latest version than maintain a custom fork of cantools. Since if we ever want to update the cantools version, we would need to manually update the fork every time.
> 
> If we wanted this sooner, it probably wouldn't be too difficult to adapt your change into a PR for the cantools repo to implement this feature.

Good point. Since this isnt blocking, i dont mind waiting a bit. In the case where we need to make a quick change, i dont mind using the forking approach either since we will probably stick to V33.1.1 (probably wont need to update cantools)

**Comment by timbitss** [comment]
I'll test V37 with the DCM board tmmrw. If V37 doesn't break anything, then I think it's best we wait out for [issue](https://github.com/cantools/cantools/issues/393) to be resolved. Otherwise, I'll create a fork in the main repo with the changes.

**Comment by pasitlaothamatas** [comment]
lgtm

**Review by pasitlaothamatas — APPROVED** [review]

**Comment by timbitss** [comment]
I tried building with cantools V37.7 but got this error message. After an hour of trying to debug this issue, @pasitlaothamatas and I believed it wasn't worth the time. 
image

**Pull request merged**